### PR TITLE
Msig

### PR DESCRIPTION
**Pull request by jennijuju**
requires format update.

### CODE REVIEW

File: docs/get-started/lotus/multisig.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Multisig wallet
+
+Multisig(multi-signature) wallet refers to a wallet that requires multiple keys to authorize a FIL transactions.
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:3]:
```suggestion
A multisig (multi-signature) wallet refers to a wallet that requires multiple keys to authorize a `FIL` transactions.
```

File: docs/get-started/lotus/multisig.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+
+## Create a multsig wallet
+
+To create a multisig, run
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:8]:
```suggestion
Use `lotus msig create` to a multisig wallet:
```

File: docs/get-started/lotus/multisig.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+```bash
+lotus msig create signerAddress1 signerAddress2 signerAddress3...
+```
+
+When a multisig wallet is successfully created, the output will be something like
+
+```bash
+Created new multisig:  f01002 f24mscgjtgymb3dqtm4ycwydh4nhygktgxm3nbgva
+```
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:18]:
````suggestion
```bash
lotus msig create signerAddress1 signerAddress2 signerAddress3...

> Created new multisig:  f01002 f24mscgjtgymb3dqtm4ycwydh4nhygktgxm3nbgva
```
````

File: docs/get-started/lotus/multisig.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+where `f01002` is the id address and `f24mscgjtgymb3dqtm4ycwydh4nhygktgxm3nbgva` is the actor address, both addresses represent the newly created multisig wallet.
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:20]:
```suggestion
In the above example, `f01002` is the id address and `f24mscgjtgymb3dqtm4ycwydh4nhygktgxm3nbgva` is the actor address. Both addresses represent the newly created multisig wallet.
```

File: docs/get-started/lotus/multisig.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+By default, signatures from all signers are required for approving a transaction, however, the number of required approvals can be customized by passing in the `--required` option upon creation:
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:22]:
```suggestion
By default, signatures from all signers are required for approving a transaction. However, you can change the number of required approvals by using the `--required` option:
```

File: docs/get-started/lotus/multisig.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+```bash
+lotus msig create --required=2 signerAddress1 signerAddress2 signerAddress3
+````
+
+The above command creates a multisig wallet that has 3 signers, and 2 approvals are required for a transaction to be executed.
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:28]:
```suggestion
The above example creates a multisig wallet with three signers, but only requires two approvals for the transaction to be executed.
```

File: docs/get-started/lotus/multisig.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+## Propose and approve a transaction
+
+Any signer of the multisig wallet can propose a transaction. The proposer approves the transaction upon proposal. The transaction will only be executed when the amount of approvals received equals to the amount of required approvals(see the inspect the wallet session for getting the threshold). If the multisig wallet only requires one signer, then the transaction will be executed once the transaction is proposed.
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:32]:
```suggestion
Any signer of the multisig wallet can _propose_ a transaction. The _proposer_ automatically approves the transaction upon proposal. The transaction will only be executed when the amount of approvals received equals to the amount of required approvals. A [multisig wallet can be inspected](#inspect-a-multisig-wallet) to get the number of required approvals. If the multisig wallet only requires one signer, then the transaction will be executed once the transaction is proposed.
```

File: docs/get-started/lotus/multisig.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+You can propose a transaction to a multisig wallet that you are a signer of by running
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:34]:
```suggestion
Use `lotus msig propose` to propose a transaction:
```

File: docs/get-started/lotus/multisig.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+```bash
+lotus msig propose --from=proposerAddress walletAddress destinationAddress value
+```
+
+A message CID and the transaction id will be returned:
+
+```bash
+send proposal in message:  bafy2bzaceajm2mghc5rludlbcr3bnpqgcz5m6gmldq6ycrc4trkejz36tnrqe
+Transaction ID: 0
+```
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:45]:
```suggestion
```bash
lotus msig propose --from=proposerAddress walletAddress destinationAddress value

> send proposal in message:  bafy2bzaceajm2mghc5rludlbcr3bnpqgcz5m6gmldq6ycrc4trkejz36tnrqe
Transaction ID: 0
```

File: docs/get-started/lotus/multisig.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+where the `bafy2bzaceajm2mghc5rludlbcr3bnpqgcz5m6gmldq6ycrc4trkejz36tnrqe` is the `messageID` and `0` is the transaction id.
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:47]:
```suggestion
In the above example `bafy2bzaceajm2mghc5rludlbcr3bnpqgcz5m6gmldq6ycrc4trkejz36tnrqe` is the `messageID`, and `0` is the `transactionID`.
```

File: docs/get-started/lotus/multisig.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+Other signers can approved the messages by running:
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:49]:
```suggestion
Other signers can use `lotus msig approve` to approve the messages:
```

File: docs/get-started/lotus/multisig.md [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+
+```bash
+lotus msig approve walletAddress transactionID proposerAddress destinationAddress value
+```
+
+Note that the value of `transactionID`, `proposerAddress`, `destinationAddress` and `value` must all match the proposal params.
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:55]:
```suggestion
The value of `transactionID`, `proposerAddress`, `destinationAddress` and `value` must match the values used in the proposal.
```

File: docs/get-started/lotus/multisig.md [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+## Inspect a multisig wallet
+
+To get the information of a multisig wallet, run
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:59]:
```suggestion
Use `lotus msig inspect` to get information about the multisig wallet:
```

File: docs/get-started/lotus/multisig.md [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+
+```bash
+lotus msign inspect walletaddress
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:62]:
```suggestion
lotus msig inspect walletaddress
```

@jennijuju typo?

File: docs/get-started/lotus/multisig.md [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+Threshold: 2 / 3 # amount of signature required / amount of signers the wallet has
+Signers:
+ID      Address
+t01001  t1ai2gcr2xlpyxcbjlegojbpr3ovdyfdyzigjoyza
+t0100   t3r4d3avth4agwxy6ko35reeuydcqaa6cq4mt6owg3zjq23pxqc6xvn7scb43dyhaf2cjnjhtioek6innbpgda
+t01003  t3rpukrggza4jjt6vpihiqoekth6tiopzhvxbp36qhrzfu4xpk6n3mxo5geh6bdavkkkhqk7owt2an2wrundtq
+Transactions:  1
+ID      State    Approvals  To                                         Value   Method   Params
+0       pending  1          t1fjswymsauvfh5zxw34t2pgz7iev2fn56unyw6ci  20 FIL  Send(0)
+```
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:79]:
```suggestion

> Balance: 0 FIL
> Spendable: 0 FIL
> Threshold: 2 / 3 # amount of signature required / amount of signers the wallet has
> Signers:
> ID      Address
> t01001  t1ai2gcr2xlpyxcbjlegojbpr3ovdyfdyzigjoyza
> t0100   t3r4d3avth4agwxy6ko35reeuydcqaa6cq4mt6owg3zjq23pxqc6xvn7scb43dyhaf2cjnjhtioek6innbpgda
> t01003  t3rpukrggza4jjt6vpihiqoekth6tiopzhvxbp36qhrzfu4xpk6n3mxo5geh6bdavkkkhqk7owt2an2wrundtq
> Transactions:  1
> ID      State    Approvals  To                                         Value   Method   Params
> 0       pending  1          t1fjswymsauvfh5zxw34t2pgz7iev2fn56unyw6ci  20 FIL  Send(0)
```
```

**Review by johnnymatthews — CHANGES_REQUESTED** [review]

**Review by johnnymatthews — APPROVED** [review]

**Comment by johnnymatthews** [comment]
Closes #523

File: docs/get-started/lotus/multisig.md [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+
+In the above example, `f01002` is the id address and `f24mscgjtgymb3dqtm4ycwydh4nhygktgxm3nbgva` is the actor address. Both addresses represent the newly created multisig wallet.
+
+By default, signatures from all signers are required for approving a transaction. However, you can change the number of required approvals by using the `--required` option:
+
+```bash
+lotus msig create --required=2 signerAddress1 signerAddress2 signerAddress3
+````
+
+The above example creates a multisig wallet with three signers, but only requires two approvals for the transaction to be executed.
```
Comment (arajasek) [inline docs/get-started/lotus/multisig.md:24]:
```suggestion
The above example creates a multisig wallet with three signers, but only requires two approvals for a transaction to be executed.
```

File: docs/get-started/lotus/multisig.md [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+
+## Propose and approve a transaction
+
+Any signer of the multisig wallet can _propose_ a transaction. The _proposer_ automatically approves the transaction upon proposal. The transaction will only be executed when the amount of approvals received equals to the amount of required approvals. A [multisig wallet can be inspected](#inspect-a-multisig-wallet) to get the number of required approvals. If the multisig wallet only requires one signer, then the transaction will be executed once the transaction is proposed.
```
Comment (arajasek) [inline docs/get-started/lotus/multisig.md:28]:
```suggestion
Any signer of a multisig wallet can _propose_ a transaction. The _proposer_ automatically approves a transaction upon proposal. A transaction will only be executed when the number of approvals received equals the number of required approvals. A [multisig wallet can be inspected](#inspect-a-multisig-wallet) to get the number of required approvals. If a multisig wallet only requires one signer, then a transaction will be executed immediately upon its proposal.
```

File: docs/get-started/lotus/multisig.md [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+## Inspect a multisig wallet
+
+Use `lotus msig inspect` to get information about the multisig wallet:
+
+```bash
+lotus msig inspect walletaddress
+
+> Balance: 0 FIL
+> Spendable: 0 FIL
+> Threshold: 2 / 3 # amount of signature required / amount of signers the wallet has
```
Comment (arajasek) [inline docs/get-started/lotus/multisig.md:57]:
```suggestion
> Threshold: 2 / 3 # number of signature required / number of signers the wallet has
```

**Review by arajasek — APPROVED** [review]
Some nits

File: docs/get-started/lotus/multisig.md [hunk 19]
```diff
@@ -0,0 +1,65 @@
+# Multisig wallet
+
+A multisig (multi-signature) wallet refers to a wallet that requires multiple keys to authorize a `FIL` transactions.
```
Comment (johnnymatthews) [inline docs/get-started/lotus/multisig.md:3]:
```suggestion
A multi-signature (multisig) wallet refers to a wallet that requires multiple keys to authorize a `FIL` transactions.
```

**Review by johnnymatthews — APPROVED** [review]

**Comment by johnnymatthews** [comment]
Commited @arajasek's suggestions. Looks good from my end. Merge when ready.

**Pull request merged**